NiBAx/plugins/SmileGAN/SmileGAN.py

```diff
@@ -0,0 +1,292 @@
+from PyQt5.QtGui import *
+from matplotlib.backends.backend_qt5 import FigureCanvasQT
+from PyQt5 import QtGui, QtCore, QtWidgets, uic
+import joblib
+import sys, os, time
+import seaborn as sns
+import numpy as np
+import pandas as pd
+from NiBAx.core.plotcanvas import PlotCanvas
+from NiBAx.core.baseplugin import BasePlugin
+from NiBAx.core.gui.SearchableQComboBox import SearchableQComboBox
+
+class computeSPAREs(QtWidgets.QWidget,BasePlugin):
+
+    #constructor
+    def __init__(self):
+        super(computeSPAREs,self).__init__()
+        self.model = {'BrainAge': None, 'AD': None}
+        root = os.path.dirname(__file__)
+        self.readAdditionalInformation(root)
+        self.ui = uic.loadUi(os.path.join(root, 'SmileGAN.ui'),self)
+        self.ui.comboBoxHue = SearchableQComboBox(self.ui)
+        self.ui.horizontalLayout_3.addWidget(self.comboBoxHue)
+        self.plotCanvas = PlotCanvas(self.ui.page_2)
+        self.ui.verticalLayout.addWidget(self.plotCanvas)
+        self.plotCanvas.axes = self.plotCanvas.fig.add_subplot(111)
+        self.SPAREs = None
+        self.ui.stackedWidget.setCurrentIndex(0)
+        self.ui.factorial_progressBar.setValue(0)
```

Comment:
Please change the name of progressBar. It says 'factorial_progressBar' and is confusing since there is no factorial computation.